xmake/rules/c++/modules/modules_support/builder.lua

```diff
@@ -460,3 +461,32 @@ function add_headerunit_to_target_mapper(target, headerunit, bmifile)
     return deduplicated and true or false
 end
 
+-- check if dependencies changed
+function is_dependencies_changed(target, module)
+    local cachekey = target:name() .. module.name
+
+    for required, _ in table.orderpairs(module.requires) do
+        requires = requires or hashset.new()
+        requires:insert(required)
+    end
+
+    local oldrequires = compiler_support.memcache():get2(cachekey, "oldrequires")
+                        or compiler_support.localcache():get2(cachekey, "oldrequires")
```

Comment:
why use memcache and localcache at same time? localcache will also save values to memory.

Reply:
In addition, does the flags here need to be persisted to a file?